### ISSUE TITLE
UsdStage::CreateNew fails on network mapped drive for Windows

### ISSUE THREAD

**Issue by nvidia-jomiller** [PixarAnimationStudios/USD#1759]
### Description of Issue
We have encountered an issue when calling `UsdStage::CreateNew` with a networked mapped drive where part of the path does not exist. `UsdStage::CreateNew` will fail when called with a path such as `Z:\\test_usd\\test.usd` on the first try. The folder `test_usd` will be created but the file `test.usd.` will not. Running `UsdStage::CreateNew` a couple more times with the same path will eventually succeed. Calling `UsdStage::CreateNew` with a path pointing to a local drive (`C:\\test_usd\\test.usd`) is fine.

After some initial investigation the problem seems to reside somewhere in `TfRealPath`, `TfFindLongestAccessiblePrefix` and `TfPathExists`. However, I suspect this is probably related to `GetFileAttributesW` with latency accessing the file server

### Steps to Reproduce
1. Launch a Python Interpreter (I used the one within usdview)
2. Run the following Python:
```python
from pxr import Usd

# Note that Z: is a mapped network drive to some file server
new_stage = Usd.Stage.CreateNew("Z:\\test_usd\\test.usd")

Traceback (most recent call last):
  File "<input>", line 1, in <module>
pxr.Tf.ErrorException: 
	Error in 'pxrInternal_v0_22__pxrReserved__::TfSafeOutputFile::Replace' at line 138 in file ..\pxr\base\tf\safeOutputFile.cpp : 'Unable to determine the real path for 'Z:\test_usd\test.usd': The system cannot find the path specified.
```
3. Running the same Python command again, or a couple of times, will succeed.

### System Information (OS, Hardware)
OS: Windows 10 Enterprise
CPU: AMD Ryzen 9 5950X 16-Core Processor
GPU: NVIDIA GeForce RTX 3080

### Package Versions
- Tested against [dev](https://github.com/PixarAnimationStudios/USD/commit/eb43fb585e8aacf9383280c5b48f9dce5627b524)

**Comment by Andy-Gill-EM** [comment]
Yeah, I've been encountering this as well. Very annoying with how our file management is currently set up.

**Comment by jilliene** [comment]
Filed as internal issue #USD-7179

**Comment by sunyab** [comment]
This was reported as fixed by PR #1871, which was released in v22.08. Closing this out.
